azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1774,144 +1784,177 @@ private void ajaxHandleUpload(final HttpServletRequest req, final HttpServletRes
       props.put(ValidatorConfigs.CUSTOM_AUTO_FIX_FLAG_PARAM, "true");
     }
 
+    ret.put("projectId", String.valueOf(project.getId()));
+
+    final File tempDir = Utils.createTempDir();
+    OutputStream out = null;
+    try {
+      logger.info("Uploading file " + name);
+      final File archiveFile = new File(tempDir, name);
+      out = new BufferedOutputStream(new FileOutputStream(archiveFile));
+      IOUtils.copy(item.getInputStream(), out);
+      out.close();
+
+      if (this.enableQuartz) {
+        //todo chengren311: should maintain atomicity,
+        // e.g, if uploadProject fails, associated schedule shouldn't be added.
+        this.scheduler.unschedule(project);
+      }
+
+      // get the locked flows for the project, so that they can be locked again after upload
+      final List<String> lockedFlows = getLockedFlows(project);
+
+      final Map<String, ValidationReport> reports = this.projectManager
+          .uploadProject(project, archiveFile, type, user, props);
+
+      if (this.enableQuartz) {
+        this.scheduler.schedule(project, user.getUserId());
+      }
+
+      // reset locks for flows as needed
+      lockFlowsForProject(project, lockedFlows);
+
+      // remove schedule of renamed/deleted flows
+      removeScheduleOfDeletedFlows(project, this.scheduleManager, (schedule) -> {
+        logger.info(
+            "Removed schedule with id {} of renamed/deleted flow: {} from project: {}.",
+            schedule.getScheduleId(), schedule.getFlowName(), schedule.getProjectName());
+        this.projectManager.postProjectEvent(project, EventType.SCHEDULE, "azkaban",
+            "Schedule " + schedule.toString() + " has been removed.");
+      });
+
+      registerErrorsAndWarningsFromValidationReport(resp, ret, reports);
+    } catch (final Exception e) {
+      logger.info("Installation Failed.", e);
+      String error = e.getMessage();
+      if (error.length() > 512) {
```

Comment:
Please add some constants for 512 and 500.

Reply:
Using constants for HTTP status code now, thanks.